pkg/network/vmispec/hotplug_test.go

```diff
@@ -80,135 +75,4 @@ var _ = Describe("utilitary funcs to identify attachments to hotplug", func() {
-
-			testNetworkName1 = "testnet1"
-			testNetworkName2 = "testnet2"
-			testNetworkName3 = "testnet3"
-			testNetworkName4 = "testnet4"
-		)
-		DescribeTable("calculate if changes are required",
-			func(vmi *v1.VirtualMachineInstance, pod *k8sv1.Pod, expIfaces []v1.Interface, expNets []v1.Network, expToChange bool) {
-				ifaces, nets, exists := vmispec.CalculateInterfacesAndNetworksForMultusAnnotationUpdate(vmi)
-				Expect(ifaces).To(Equal(expIfaces))
-				Expect(nets).To(Equal(expNets))
-				Expect(exists).To(Equal(expToChange))
-			},
-			Entry("when no interfaces exist, change is not required", libvmi.New(), nil, nil, nil, expectNoChange),
-			Entry("when vmi interfaces match pod multus annotation and status, change is not required",
-				libvmi.New(
-					libvmi.WithInterface(v1.Interface{Name: testNetworkName1}),
-					libvmi.WithNetwork(&v1.Network{Name: testNetworkName1}),
-					withInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{Name: testNetworkName1}),
-				),
-				&k8sv1.Pod{
-					ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{
-						networkv1.NetworkStatusAnnot: `[
-						{"interface":"net1", "name":"red-net", "namespace": "default"}
-					]`,
-					}},
-				}, nil, nil, expectNoChange,
-			),
-			Entry("when vmi interfaces have an extra interface which requires hotplug",
-				libvmi.New(
-					libvmi.WithInterface(v1.Interface{Name: testNetworkName1, InterfaceBindingMethod: v1.InterfaceBindingMethod{SRIOV: &v1.InterfaceSRIOV{}}}),
-					libvmi.WithInterface(v1.Interface{Name: testNetworkName2}),
-					libvmi.WithInterface(v1.Interface{Name: testNetworkName3}),
-					libvmi.WithInterface(v1.Interface{Name: testNetworkName4, InterfaceBindingMethod: v1.InterfaceBindingMethod{SRIOV: &v1.InterfaceSRIOV{}}}),
-					libvmi.WithNetwork(&v1.Network{Name: testNetworkName1}),
-					libvmi.WithNetwork(&v1.Network{Name: testNetworkName2}),
-					libvmi.WithNetwork(&v1.Network{Name: testNetworkName3}),
-					libvmi.WithNetwork(&v1.Network{Name: testNetworkName4}),
-					withInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{Name: testNetworkName1}),
-					withInterfaceStatus(v1.VirtualMachineInstanceNetworkInterface{Name: testNetworkName2}),
-				),
-				&k8sv1.Pod{
-					ObjectMeta: metav1.ObjectMeta{Annotations: map[string]string{
-						networkv1.NetworkStatusAnnot: `[
-						{"interface":"net1", "name":"red-net", "namespace": "default"},
-						{"interface":"net2", "name":"blue-net", "namespace": "default"}
-					]`,
-					}},
-				},
-				[]v1.Interface{{Name: testNetworkName1, InterfaceBindingMethod: v1.InterfaceBindingMethod{SRIOV: &v1.InterfaceSRIOV{}}}, {Name: testNetworkName2}, {Name: testNetworkName3}},
-				[]v1.Network{{Name: testNetworkName1}, {Name: testNetworkName2}, {Name: testNetworkName3}},
-				expectToChange,
-			),
-			Entry("when vmi interfaces have an extra SRIOV interface which requires hotplug, change is not required since SRIOV hotplug to a pod is not supported",
```

Comment:
I think you added explicitly this scenario in the prev commit to the caller function on which you depend now. But you have reduced it a bit, why? In retro, this move/conversion should have been done in one commit to emphasize what moved (e.g. in the prev commit, to delete this test from here).

Anyway, as a reviewer, I am suppose to check in depth that all the other tests here, with all their details are covered at the caller. If you already done that, please provide the relevant ref to help me as well.

If it was not done, then I would expect all these cases to be exercised on the caller if possible. If not possible, then mention which ones and why.

Reply:
>  In retro, this move/conversion should have been done in one commit to emphasize what moved (e.g. in the prev commit, to delete this test from here).

Done.
Same for the other tests.

> Anyway, as a reviewer, I am suppose to check in depth that all the other tests here, with all their details are covered at the caller. If you already done that, please provide the relevant ref to help me as well.

If it was not done, then I would expect all these cases to be exercised on the caller if possible. If not possible, then mention which ones and why.

Done.